src/whylogs/core/datasetprofile.py

```diff
@@ -282,6 +282,7 @@ def track_dataframe(self, df: pd.DataFrame, character_list=None, token_method=No
         # workaround for CUDF due to https://github.com/rapidsai/cudf/issues/6743
         if cudfDataFrame is not None and isinstance(df, cudfDataFrame):
             df = df.to_pandas()
+        self.total_row_number = len(df.index)
```

Comment:
The way it was implemented, when logging sequentially it would have been a problem since the total_row_number gets overwritten.
And after taking a closer look, there are functions that use track() and/or track_datum() methods directly which would be a problem as well. 
My bad I didn't realize those things earlier.
What I changed, as every method used for tracking (track_dataframe(), track(), track_array() etc.) in the DatasetProfile, ends up using the track_datum() method, that's where I update the variables so everything that gets logged is being accounted for.